src/com/google/enterprise/adaptor/prebuilt/SkipDocumentFilter.java

```diff
@@ -142,7 +144,15 @@ private SkipDocumentFilter(String propertyName, Pattern pattern,
    */
   private boolean foundInMetadata(Metadata metadata) {
     boolean found = false;
-    for (String value : metadata.getAllValues(propertyName)) {
+    Set<String> values = metadata.getAllValues(propertyName);
+    if (values.isEmpty() && log.isLoggable(Level.FINEST)) {
+      if (metadata.getKeys().contains(propertyName)) {
+        log.finest("No values for key `" + propertyName + "' in metadata.");
```

Comment:
The use of ` feels like an affectation, and is used only 8 times in the full googlegsa code base (compared to 141 occurrences of '"). Let's not add 6 more. If we takeout the existing two each in SkipDocumentFilter and RegexDocumentFilter, then the library would have zero.